### DISCUSSION TITLE
How to add some distortion effect for camera output

### DISCUSSION THREAD

**Discussion by bappum** [google/filament#7807] [Q&A]
I am quite new to this filament and 3d scenes please help with pointers any. 

Say I have a scene with various nodes in it. and I wanted some distortion effect (pincushion mostly) to be added to the camera output.

**Comment by bappum** [comment]
any directions/suggestions/ relevant examples would be greatly helpful

**Comment by Nolram12345** [comment]
As I was told in [this discussion](https://github.com/google/filament/discussions/7676), there isn't currently an official API for adding new post-processing effects to Filament - and it doesn't seem like something that's going to be added by the team in the near future anyways, at least at the moment (maybe if you open an issue for it and other people show support it might become a higher priority for them?). 

So right now, the only option I'm aware of would be to hack filament itself to add a custom post processing effect. I'm currently trying to figure out how to do this properly. Romainguy also mentioned it may be possible with off-screen render textures, but never elaborated on that unfortunately.
